packages/scope-sdk/src/Scope.ts

```diff
@@ -448,15 +490,63 @@ export class Scope {
       commitment: this._connection.commitment,
     });
     const mappings = await this.getOracleMappings(feed);
-    tokens.forEach((token) => {
-      refreshIx.keys.push({
-        isSigner: false,
-        isWritable: false,
-        pubkey: mappings.priceInfoAccounts[token],
-      });
-    });
+    for (const token of tokens) {
+      refreshIx.keys.push(
+        ...(await Scope.getRefreshAccounts(this._connection, this._config.kamino.programId, mappings, token))
+      );
+    }
     return provider.send(new Transaction().add(refreshIx), [payer]);
   }
+
+  static async getRefreshAccounts(
+    connection: Connection,
+    kaminoProgramId: PublicKey,
+    mappings: OracleMappings,
+    token: number
+  ): Promise<AccountMeta[]> {
+    const keys: AccountMeta[] = [];
+    keys.push({
+      isSigner: false,
+      isWritable: false,
+      pubkey: mappings.priceInfoAccounts[token],
+    });
+    switch (mappings.priceTypes[token]) {
+      case new OracleType.KToken().discriminator:
+        keys.push(...(await Scope.getKTokenRefreshAccounts(connection, kaminoProgramId, mappings, token)));
```

Comment:
Nitpick - I would preemptively add a break inside this case statement, just in case we ever another branch to the switch statement, it could cause some unexpected problems down the road.